nixos/modules/services/web-servers/nginx/default.nix

```
@@ -499,13 +519,31 @@ in
         '';
       };
 
+      defaultListen = mkOption {
+        type = with types; listOf (submodule { options = {
+         addr = mkOption { type = str;  description = lib.mdDoc "IP address.";  };
+         port = mkOption { type = nullOr port;  description = lib.mdDoc "Port number."; default = null; };
+         ssl  = mkOption { type = nullOr bool; description = lib.mdDoc "Enable SSL."; default = null; };
+         proxyProtocol = mkOption { type = nullOr bool; description = lib.mdDoc "Enable PROXY protocol."; default = null; };
+         extraParameters = mkOption { type = nullOr (listOf str); description = lib.mdDoc "Extra parameters of this listen directive."; default = null; example = [ "backlog=1024" "deferred" ]; };
+        }; });
```

```suggestion
        type = with types; listOf (submodule {
          options = {
            addr = mkOption {
              type = str;
              description = lib.mdDoc "IP address.";
            };
            port = mkOption {
              type = nullOr port;
              description = lib.mdDoc "Port number.";
              default = null;
            };
            ssl  = mkOption {
              type = nullOr bool;
              default = null;
              description = lib.mdDoc "Enable SSL.";
            };
            proxyProtocol = mkOption {
              type = nullOr bool;
              description = lib.mdDoc "Enable PROXY protocol.";
              default = null;
            };
            extraParameters = mkOption {
              type = nullOr (listOf str);
              description = lib.mdDoc "Extra parameters of this listen directive.";
              default = null;
              example = [ "backlog=1024" "deferred" ];
            };
          };
        });
```
proxyProtocoll and ssl are bool types, why not default them to false?

---

`ssl = null;` means that you allow whatever makes sense at generation time.
whereas `ssl = false;` no SSL no matter what or `ssl = true;` SSL only

For `proxyProtocol`, it could indeed makes sense to have it as a boolean defaulting to false. Checking if this is possible.

---

Addressed.